tests/helpers/perf_results_helper.rb

```diff
@@ -120,21 +105,90 @@ def gatling2csv(results_dir)
     csv2html(output_path)
   end
 
+  # Parse the stats.json file and return the group node
+  #
+  # @author Bill Claytor
+  #
+  # @param [String] stats_path The path to the stats.json file
+  #
+  # @raise [StandardError] If the specified file is not found
+  # @raise [StandardError] If the specified file does not contain the group node
+  #
+  # @return [JSON] The group node
+  #
+  # @example
+  #   group_node = gatling_json_stats_group_node(stats_path)
+  #
+  def gatling_json_stats_group_node(stats_path)
+    raise "The specified file was not found: #{stats_path}" unless File.exist?(stats_path)
+
+    puts "Examining Gatling data: #{stats_path}"
+    puts
+
+    begin
+      # the 'group' name will be something like 'group_nooptestwithout-9eb19'
+      stats = JSON.parse(File.open(stats_path).read)
+      group_keys = stats["contents"].keys.select { |key| key.to_s.match(/group/) }
+      group_node = stats["contents"][group_keys[0]]
+    rescue StandardError
+      raise "JSON parse of #{stats_path} should result in one key matching 'group'"
+    end
+
+    group_node
+  end
+
+  # Retrieves the contents from the group node
+  #
+  # @author Bill Claytor
+  #
+  # @param [JSON] The group node
+  #
+  # @raise [StandardError] If the group node does not contain contents with more than one key
+  #
+  # @return [JSON] The contents
+  #
+  # @example
+  #   contents = gatling_json_stats_group_node_contents(group_node)
+  #
+  def gatling_json_stats_group_node_contents(group_node)
+    contents = group_node["contents"]
+
+    if contents.nil? || contents.empty? || contents.keys.empty?
+      raise "The 'contents' element of the 'group' node must have at least one key"
+    end
+
+    # TODO: verify each key
+    # TODO: unit test to ensure data validity
+    puts "There are #{contents.keys.length} keys"
+    puts
+
+    (0..contents.keys.length - 1).each do |i|
```

Comment:
Why is the last key skipped?

Reply:
This is actually older code that was extracted. After adding tests to confirm that all keys are included I started to update it to just use the keys directly but realized that the intention was to reference the keys by index:
```
puts "key #{i}: #{name}")
```
I think this was initially added for debugging; not sure if it is still useful.